sormas-ui/src/main/java/de/symeda/sormas/ui/utils/CommitDiscardWrapperComponent.java

```diff
@@ -979,6 +984,31 @@ public void setDirty(boolean dirty) {
 		this.dirty = dirty;
 	}
 
+	//In case of having delete right without edit right the delete button should remain enabled
+	public void restrictEditableComponentsOnEditView(UserRight editRight, UserRight deleteRight, EditPermissionType editPermissionType) {
+		boolean isEditAllowed = isEditAllowed(editRight, editPermissionType);
+		if (!isEditAllowed) {
+			if (isDeleteAllowed(deleteRight)) {
+				addButtonToExcludedList(CommitDiscardWrapperComponent.DELETE_UNDELETE);
+				this.setEditable(false, getExcludedButtons().stream().toArray(String[]::new));
+			} else {
+				this.setEditable(false);
+			}
+		}
+	}
+
+	public boolean isDeleteAllowed(UserRight deleteRight) {
+		return UserProvider.getCurrent().hasUserRight(deleteRight);
+	}
+
+	public boolean isEditAllowed(UserRight editRight, EditPermissionType editPermissionType) {
+		if (editPermissionType != null) {
```

Comment:
this can be simplified to 
```
UserProvider.getCurrent().hasUserRight(editRight) && (editPermissionType == null || editPermissionType == EditPermissionType.ALLOWED)
```